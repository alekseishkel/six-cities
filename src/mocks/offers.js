const offersCardsInfo = [{
  description: `Beautiful & luxurious apartment at great location`,
  img: `img/apartment-01.jpg`,
  price: `€120`,
  type: `Apartment`,
  isPremium: true
},
{
  description: `Wood and stone place`,
  img: `img/room.jpg`,
  price: `€80`,
  type: `Private room`,
  isPremium: false
},
{
  description: `Canal View Prinsengracht`,
  img: `img/apartment-02.jpg`,
  price: `€132`,
  type: `Apartment`,
  isPremium: false
},
{
  description: `Nice, cozy, warm big bed apartment`,
  img: `img/apartment-03.jpg`,
  price: `€180`,
  type: `Apartment`,
  isPremium: false
}];

export default offersCardsInfo;
