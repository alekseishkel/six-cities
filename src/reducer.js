const ActionCreator = {
  changeCity: (city) => ({
    type: `CHANGE_CITY`,
    payload: city
  }),

  changeCityCoords: (cityCoords) => ({
    type: `CHANGE_CITY_COORDS`,
    payload: cityCoords
  }),

  changeZoom: (zoom) => ({
    type: `CHANGE_ZOOM`,
    payload: zoom
  })
};

const initialState = {
  city: `Paris`,
  cityCoords: [48.8680266086781, 2.351173200195321],
  mapZoom: 11,
  offersCoords: [
    [52.3909553943508, 4.85309666406198],
    [52.369553943508, 4.85309666406198],
    [52.3909553943508, 4.929309666406198],
    [52.3809553943508, 4.939309666406198]
  ],
  offers: [{
    description: `Beautiful & luxurious apartment at great location`,
    img: `img/apartment-01.jpg`,
    price: `€120`,
    type: `Apartment`,
    isPremium: true,
    isBookmark: false
  },
  {
    description: `Wood and stone place`,
    img: `img/room.jpg`,
    price: `€80`,
    type: `Private room`,
    isPremium: false,
    isBookmark: false
  },
  {
    description: `Canal View Prinsengracht`,
    img: `img/apartment-02.jpg`,
    price: `€132`,
    type: `Apartment`,
    isPremium: false,
    isBookmark: false
  },
  {
    description: `Nice, cozy, warm big bed apartment`,
    img: `img/apartment-03.jpg`,
    price: `€180`,
    type: `Apartment`,
    isPremium: false,
    isBookmark: false
  }]
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case `CHANGE_CITY`:
      return Object.assign({}, state, {
        city: action.payload
      });
    case `CHANGE_CITY_COORDS`:
      return Object.assign({}, state, {
        cityCoords: action.payload
      });
    case `CHANGE_ZOOM`:
      return Object.assign({}, state, {
        mapZoom: action.payload
      });
    default:
      return state;
  }
};

export {
  ActionCreator,
  reducer
};
