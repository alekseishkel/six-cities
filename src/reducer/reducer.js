const ActionCreator = {
  changeCity: (city) => ({
    type: `CHANGE_CITY`,
    payload: city
  }),

  changeCityCoords: (cityCoords) => ({
    type: `CHANGE_CITY_COORDS`,
    payload: cityCoords
  }),

  changeZoom: (zoom) => ({
    type: `CHANGE_ZOOM`,
    payload: zoom
  })
};

const initialState = {
  city: `Paris`,
  cityCoords: [48.8680266086781, 2.351173200195321],
  mapZoom: 11,
};

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case `CHANGE_CITY`:
      return Object.assign({}, state, {
        city: action.payload
      });
    case `CHANGE_CITY_COORDS`:
      return Object.assign({}, state, {
        cityCoords: action.payload
      });
    case `CHANGE_ZOOM`:
      return Object.assign({}, state, {
        mapZoom: action.payload
      });
    default:
      return state;
  }
};

export {
  ActionCreator,
  reducer
};
