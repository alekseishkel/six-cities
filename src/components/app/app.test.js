import React from 'react';
import renderer from 'react-test-renderer';

import {createStore} from 'redux';
import {Provider} from 'react-redux';
import reducer from '../../reducer/reducer';

import App from './app';

import offers from '../../mocks/offers';

it(`App correctly renders after relaunch`, () => {
  const store = createStore(reducer);

  const mapContainer = document.createElement(`mapContainer`);
  mapContainer.setAttribute(`id`, `map`);
  document.body.appendChild(mapContainer);

  const tree = renderer.create(
      <Provider store={store}>
        <App
          currentCity={`Cologne`}
          offers={offers}
        />
      </Provider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
