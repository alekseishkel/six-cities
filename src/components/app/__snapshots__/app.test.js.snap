// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App correctly renders after relaunch 1`] = `
<div
  className="lds-ripple"
>
  <div />
  <div />
</div>
`;
